pulp_gem/app/tasks/synchronizing.py

```diff
@@ -72,54 +73,86 @@ async def run(self):
-                        versions = [
-                            version for version in versions if VERSION_REGEX.fullmatch(version)
-                        ]
+                    if includes is not None:
+                        if name not in includes:
+                            continue
+                        include_versions = includes[name]
                     else:
-                        versions = [
-                            version
-                            for version in versions
-                            if PRERELEASE_VERSION_REGEX.fullmatch(version)
-                        ]
-                    if self.remote.includes:
-                        if name not in self.remote.includes:
+                        include_versions = None
+                    if excludes is not None and name in excludes:
+                        exclude_versions = excludes[name]
+                        if exclude_versions is None:
                             continue
-                        version_requirements = self.remote.includes[name]
-                        if version_requirements is not None:
-                            versions = [
-                                version
-                                for version in versions
-                                if ruby_ver_includes(version_requirements, version)
-                            ]
-                    if self.remote.excludes:
-                        if name in self.remote.excludes:
-                            version_requirements = self.remote.excludes[name]
-                            if version_requirements is None:
-                                continue
-                            versions = [
-                                version
-                                for version in versions
-                                if not ruby_ver_includes(version_requirements, version)
-                            ]
-                    if not versions:
+                    else:
+                        exclude_versions = None
+
+                    # Skip conditions based on the gem version
+                    # ========================================
+
+                    # The list 'ext_versions' contains "{version}[-{platform}]" entries!
+                    # This dict is like a set of ext_versions with payload dict on
+                    # {version, platform, prerelease}.
+                    versions_info = {
+                        ext_version: split_ext_version(ext_version) for ext_version in ext_versions
+                    }
+                    if not prereleases:
+                        # Prerelease versions are already sanitized.
+                        versions_info = {
+                            k: v for k, v in versions_info.items() if not v["prerelease"]
+                        }
+                    else:
+                        # Sanitize versions.
+                        # TODO Logging?
```

Comment:
Do you want to add logging for the versions that get filtered out (the invalid ones)?